### ISSUE TITLE
PhantomJS error handling

### ISSUE THREAD

**Issue by jonbri** [ariya/phantomjs#13403]
I'm having a hard time understanding how PhantomJS handles errors.

I have a locally installed Apache server running (xampp), and when I manually visit "http://localhost/" I get the "It Works!" page.

As a test, I wrote a small file (called forceError.js) that purposely causes an unchecked exception:

```
var page = require('webpage').create(),
    url = 'http://localhost/';

page.onError = function(msg, trace) {
  console.log("page.onError");
  var msgStack = ['ERROR: ' + msg];
  if (trace && trace.length) {
    msgStack.push('TRACE:');
    trace.forEach(function(t) {
      msgStack.push(' -> ' + t.file + ': ' + t.line + (t.function ? ' (in function "' + t.function +'")' : ''));
    });
  }
  console.error(msgStack.join('\n'));
};

phantom.onError = function(msg, trace) {
  console.log("phantom.onError");
  var msgStack = ['PHANTOM ERROR: ' + msg];
  if (trace && trace.length) {
    msgStack.push('TRACE:');
    trace.forEach(function(t) {
      msgStack.push(' -> ' + (t.file || t.sourceURL) + ': ' + t.line + (t.function ? ' (in function ' + t.function +')' : ''));
    });
  }
  console.error(msgStack.join('\n'));
  phantom.exit(1);
};

page.open(url, function (status) {
    console.log("status: " + status);

    // an undefined function
    thisShouldForceAnError();
});
```

When I run this using:

```
phantomjs.exe forceError.js
```

First I get "status: success" and then the process just hangs. I don't see either page.onError or phantom.onError being invoked.

Is there some property or something I need to turn on to get general error handling?

I'm on Windows 7, PhantomJS version 2.0.0, and running this in my "git bash" shell.

Note: this question was originally posted on StackOverflow: http://stackoverflow.com/questions/31322029/phantomjs-error-handling

**Comment by bharatraj88** [comment]
Have any one found a way to over come the above issue? Am facing hard time as well while throwing an Error inside page.open block

**Comment by sryze** [comment]
I spent a good amount of time trying to guess out why my PhantomJS script was hanging and it turned out to be a simply syntax error inside my `page.open` callback, all because `phantom.onError` isn't ever called on error. I think it's a pretty serious issue, it's surprising how it hasn't been fixed yet after 1.5 years.

**Comment by RRorg** [comment]
Any updates on this?
